core/controllers/base_test.py

```diff
@@ -720,6 +721,53 @@ def test_get(self):
         self.assertEqual(call_counter.times_called, 1)
 
 
+class SessionEndHandlerTests(test_utils.GenericTestBase):
+    """Tests for session end handler."""
+
+    def test_get(self):
+        call_counter = test_utils.CallCounter(lambda *_: None)
+
+        with self.swap(auth_services, 'destroy_auth_session', call_counter):
+            self.get_html_response('/session_end', expected_status_int=200)
+
+        self.assertEqual(call_counter.times_called, 1)
+
+
+class CreateInitialSuperAdminTests(test_utils.GenericTestBase):
```

Comment:
Suggest also adding tests to verify that the redirection successfully happens both in success & failure cases (per the implementation).

Reply:
Done